src/Common/Core/Impl/Extensions/IOExtensions.cs

```diff
@@ -22,6 +23,20 @@ public static class IOExtensions {
             return path.StartsWithIgnoreCase(bp) ? path.Substring(bp.Length) : path;
         }
 
+        // TODO: merge with above
+        public static string MakeCompleteRelativePath(this string path, string basePath) {
```

Comment:
Not too happy with the name, but don't want to change behaviour of the existing `MakeRelativePath`